core/src/sumeragi/network_topology.rs

```diff
@@ -163,16 +262,39 @@ pub enum Role {
     ProxyTail,
 }
 
+/// Error during signature verification
+#[derive(thiserror::Error, Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SignatureVerificationError {
+    /// Error when block doesn't have enough valid signatures
+    #[error("The block doesn't have enough valid signatures to be committed ({votes_count} out of {min_votes_for_commit}).")]
```

Comment:
I was thinking of adding some primitive error types. We have common patterns like not enough of X, which has verbatim the same error message, that is inconsistently replicated all over the codebase. 

```rust
struct Insufficient<T: Ord> {
    expected: T, 
    actual: T, 
}
```
with some basic convenience functions.

Reply:
Created #3524.